p2p/stream/types/stream.go

```diff
@@ -82,7 +84,11 @@ func (st *BaseStream) FailedTimes() int {
 	return st.failedTimes
 }
 
-func (st *BaseStream) AddFailedTimes() {
+func (st *BaseStream) AddFailedTimes(faultRecoveryThreshold time.Duration) {
+	durationSinceLastFailure := time.Now().Sub(st.lastFailureTime)
+	if durationSinceLastFailure >= faultRecoveryThreshold {
```

Comment:
since this function is only called by `StreamFailed`, it looks to me that a failed stream will never be back to 0, and a previously failed stream will always be at 1 at the minimum until the next node restart

Reply:
They never back to 0 except if they can work without issue for 30 minutes. Otherwise once they reach max failures, they will be removed

Reply:
thanks yeah this is what was also expected as a behavior here.